core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt

```diff
@@ -229,7 +229,10 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
     }
     activeFragments().filterIsInstance<FragmentActivityExtensions>().forEach {
       if (it.onBackPressed(this) == FragmentActivityExtensions.Super.ShouldCall) {
-        if (navController.currentDestination?.label?.equals("Reader") == true) {
+        if (navController.currentDestination?.label?.equals("Reader") == true &&
+          navController.previousBackStackEntry?.destination
+            ?.label?.equals("SearchFragment") == false
```

Comment:
We should use `id` here. Hardcoded `string` is not a good idea.

Reply:
Done